lib/versioning/ivy/parse.js

```diff
@@ -0,0 +1,46 @@
+const { isSingleVersion, parseRange, rangeToStr } = require('../maven/compare');
+
+const REV_TYPE_LATEST = 'REV_TYPE_LATEST';
+const REV_TYPE_SUBREV = 'REV_TYPE_SUBREVISION';
+const REV_TYPE_RANGE = 'REV_TYPE_RANGE';
+
+function parseDynamicRevision(str) {
+  if (!str) return null;
+
+  const LATEST_REGEX = /^latest\.|^latest$/i;
+  if (LATEST_REGEX.test(str)) {
+    const value = str.replace(LATEST_REGEX, '').toLowerCase() || null;
+    return {
+      type: REV_TYPE_LATEST,
+      value: value !== 'integration' ? value : null,
+    };
+  }
+
+  const SUBREV_REGEX = /\.\+$/;
+  if (SUBREV_REGEX.test(str)) {
+    const value = str.replace(SUBREV_REGEX, '');
+    if (isSingleVersion(value)) {
+      return {
+        type: REV_TYPE_SUBREV,
+        value,
+      };
+    }
+  }
+
+  const range = parseRange(str.replace(/^]/, '(').replace(/\[$/, ')'));
```

Comment:
@rarkins Here i'm extending the range syntax, so it can be reused

Reply:
Sounds OK if you found a way to map them back and forth

Reply:
You mean, to preserve one or another style after extending? Yeah, that requires more than just couple replacings